distros/spring/src/test/java/com/connexta/transformation/service/TransformControllerTest.java

```diff
@@ -13,51 +13,90 @@
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.MediaType;
 import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
 import org.springframework.web.client.ResourceAccessException;
 
 @RunWith(SpringRunner.class)
-@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
-@TestPropertySource(properties = "endpoints.lookupService.url = http://localhost:")
-@ContextConfiguration
+@WebMvcTest(TransformController.class)
 public class TransformControllerTest {
-  private static final String ACCEPT_VERSION = "0.1.0";
 
-  @LocalServerPort private int port;
+  private static final String TEST_URI = "http://test:9000";
 
-  @Autowired private TestRestTemplate restTemplate;
+  private static final String ACCEPT_VERSION = "Accept-Version";
 
-  @Autowired private String lookupServiceUrl;
+  private static final String ACCEPT_VERSION_NUM = "0.1.0";
 
-  @Test
-  public void testAcceptedResponseEntity() {
-    TransformController controller =
-        new TransformController(restTemplate.getRestTemplate(), lookupServiceUrl + port);
-    ResponseEntity responseEntity = controller.transform(ACCEPT_VERSION, new TransformRequest());
+  private ObjectMapper mapper = new ObjectMapper();
+
+  @Autowired private MockMvc mockMvc;
 
-    Assert.assertThat(responseEntity.getStatusCode(), is(HttpStatus.ACCEPTED));
+  @MockBean TransformationManager tManager;
+
+  @MockBean(answer = Answers.RETURNS_MOCKS)
+  private RestTemplateBuilder restTemplateBuilder;
+
+  @Test
+  public void testAcceptedResponseEntity() throws Exception {
+    this.mockMvc
+        .perform(
+            post("/transform")
+                .header(ACCEPT_VERSION, ACCEPT_VERSION_NUM)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(
+                    mapper.writeValueAsString(
+                        new TransformRequest()
+                            .currentLocation(new URI(TEST_URI))
+                            .finalLocation(new URI(TEST_URI))
+                            .metacardLocation(new URI(TEST_URI))))
+                .characterEncoding(StandardCharsets.UTF_8.name()))
+        .andExpect(status().isAccepted());
   }
 
   @Test(expected = ResourceAccessException.class)
   public void testInvalidServiceUrl() {
     String badServiceUrl = "http://bad.url";
     TransformController controller =
-        new TransformController(restTemplate.getRestTemplate(), badServiceUrl);
+        new TransformController(new RestTemplateBuilder(), badServiceUrl, tManager);
 
     controller.transform(ACCEPT_VERSION, new TransformRequest());
   }
+
+  @Test
+  public void testDelete() throws Exception {
+    this.mockMvc.perform(delete("/transform/id123")).andExpect(status().isNoContent());
+    verify(tManager).delete(anyString());
```

Comment:
❗️ We should verify a lot more than just the Http status. The API specification indicates everything that is expecte to be returned in headers; these should be verified as well.

Reply:
☝️ The api only returns the status and content-version